### PR TITLE
Update Helm/Operator deployments to use 1.1.0 images and doc updates.

### PR DESCRIPTION
**Pull request by shivamerla**
Signed-off-by: Shiva Krishna, Merla <shivakrishna.merla@hpe.com>

### CODE REVIEW

File: helm/charts/hpe-csi-driver/README.md [hunk 1]
```diff
@@ -43,25 +42,57 @@ It's recommended to create a [values.yaml](https://github.com/hpe-storage/co-dep
...
+# helm v3
+helm uninstall hpe-csi
+
+# helm 2.x
 helm delete hpe-csi --purge
 ```
 
 ### Alternative install method
-In some cases it's more practical to provide the local configuration via the `helm` command directly. Specify each parameter using the `--set key=value[,key=value]` argument to `helm install`. For example:
+In some cases it's more practical to provide the local configuration via the `helm` command directly. Specify each parameter using the `--set key=value[,key=value]` argument to `helm install`. These will take precedence than entries in values.yaml. For example:
```
Comment (rgcostea) [inline helm/charts/hpe-csi-driver/README.md]: nit: take precedence over entries

Comment (shivamerla) [reply helm/charts/hpe-csi-driver/README.md]: fixed. thanks.

File: helm/charts/hpe-csi-driver/README.md [hunk 2]
```diff
@@ -6,10 +6,9 @@ The [HPE CSI Driver for Kubernetes](https://github.com/hpe-storage/csi-driver) l
 
 - Upstream Kubernetes version > 1.13
 - Other Kubernetes distributions supported
-- OpenShift 4.2 in Tech Preview (3.x is not supported, see [FlexVolume Driver Helm chart](https://github.com/hpe-storage/co-deployments/tree/master/helm/charts/hpe-flexvolume-driver))
-- More distributions will be listed as tests are ongoing
+- OpenShift 4.2, 4.3(RHCOS, RHEL)
 - Recent Ubuntu, CentOS or RHEL compute nodes connected to their respective official package repositories
-- Helm 3.x is not supported with v1.0.0 release.
+- Helm (v3 is supported only from HPE CSI Driver version 1.1.0 onwards)
```
Comment (drajen) [inline helm/charts/hpe-csi-driver/README.md]: Can't we enumerate it like this?
- Helm 2 (whatever caveat if there is one)
- Helm 3 (whatever caveat if there is one)

File: helm/charts/hpe-csi-driver/README.md [hunk 3]
```diff
@@ -6,10 +6,9 @@ The [HPE CSI Driver for Kubernetes](https://github.com/hpe-storage/csi-driver) l
 
 - Upstream Kubernetes version > 1.13
 - Other Kubernetes distributions supported
-- OpenShift 4.2 in Tech Preview (3.x is not supported, see [FlexVolume Driver Helm chart](https://github.com/hpe-storage/co-deployments/tree/master/helm/charts/hpe-flexvolume-driver))
-- More distributions will be listed as tests are ongoing
+- OpenShift 4.2, 4.3(RHCOS, RHEL)
```
Comment (drajen) [inline helm/charts/hpe-csi-driver/README.md]: - Red Hat OpenShift 4.2, 4.3 (RHCOS and RHEL)

File: helm/charts/hpe-csi-driver/README.md [hunk 4]
```diff
@@ -6,10 +6,9 @@ The [HPE CSI Driver for Kubernetes](https://github.com/hpe-storage/csi-driver) l
 
 - Upstream Kubernetes version > 1.13
 - Other Kubernetes distributions supported
```
Comment (drajen) [inline helm/charts/hpe-csi-driver/README.md]: - Most Kubernetes distributions are supported

File: helm/charts/hpe-csi-driver/README.md [hunk 5]
```diff
@@ -43,25 +42,57 @@ It's recommended to create a [values.yaml](https://github.com/hpe-storage/co-dep
 ### Installing the Chart
 To install the chart with the name `hpe-csi`:
 
+Add HPE helm repo:
 ```
 helm repo add hpe https://hpe-storage.github.io/co-deployments
-helm install --name hpe-csi hpe/hpe-csi-driver --namespace kube-system -f myvalues.yaml
+helm repo update
+helm search repo hpe-csi-driver
```
Comment (drajen) [inline helm/charts/hpe-csi-driver/README.md]: Why would you do this? (helm search)

File: helm/charts/hpe-csi-driver/README.md [hunk 6]
```diff
@@ -43,25 +42,57 @@ It's recommended to create a [values.yaml](https://github.com/hpe-storage/co-dep
...
 ```
 
-**Note:** Omitting the `--name` flag will generate a human readable name.
+Install the latest chart:
+
+```
+# helm v3
```
Comment (drajen) [inline helm/charts/hpe-csi-driver/README.md]: The official Helm nomenclature is to use "Helm 2" and "Helm 3" respectively.

File: helm/charts/hpe-csi-driver/README.md [hunk 7]
```diff
@@ -43,25 +42,57 @@ It's recommended to create a [values.yaml](https://github.com/hpe-storage/co-dep
...
+helm install hpe-csi hpe/hpe-csi-driver --namespace kube-system -f myvalues.yaml
+
+# helm 2.x
```
Comment (drajen) [inline helm/charts/hpe-csi-driver/README.md]: See above.

File: helm/charts/hpe-csi-driver/README.md [hunk 8]
```diff
@@ -43,25 +42,57 @@ It's recommended to create a [values.yaml](https://github.com/hpe-storage/co-dep
...
+helm install --name hpe-csi hpe/hpe-csi-driver --namespace kube-system -f myvalues.yaml --set crd.nodeInfo.create=true
+```
+
+### Upgrading the Chart
+To upgrade the chart, specify the version you want to upgrade to as below. Please do NOT re-use the values.yaml from 1.0.0 version to upgrade to later versions. Always use values.yaml from corresponding release from [values.yaml](https://github.com/hpe-storage/co-deployments/blob/master/helm/values/csi-driver)
+
+```
+# list the avaiable version of the plugin:
+helm repo update
+helm search repo hpe-csi-driver -l
```
Comment (drajen) [inline helm/charts/hpe-csi-driver/README.md]: What is this search business?

File: helm/charts/hpe-csi-driver/README.md [hunk 9]
```diff
@@ -43,25 +42,57 @@ It's recommended to create a [values.yaml](https://github.com/hpe-storage/co-dep
...
+# select the target version to upgrade as below:
+helm upgrade hpe-csi hpe/hpe-csi-driver --namespace kube-system --version=x.x.x.x -f myvalues.yaml
+```
 
 ### Uninstalling the Chart
-To uninstall/delete the `hpe-csi` deployment:
+To uninstall/delete the `hpe-csi` chart:
 
 ```
+# helm v3
```
Comment (drajen) [inline helm/charts/hpe-csi-driver/README.md]: Helm 3

File: helm/charts/hpe-csi-driver/README.md [hunk 10]
```diff
@@ -43,25 +42,57 @@ It's recommended to create a [values.yaml](https://github.com/hpe-storage/co-dep
...
+helm uninstall hpe-csi
+
+# helm 2.x
```
Comment (drajen) [inline helm/charts/hpe-csi-driver/README.md]: Helm 2

File: helm/charts/hpe-csi-driver/README.md [hunk 11]
```diff
@@ -43,25 +42,57 @@ It's recommended to create a [values.yaml](https://github.com/hpe-storage/co-dep
...
 helm delete hpe-csi --purge
 ```
 
 ### Alternative install method
-In some cases it's more practical to provide the local configuration via the `helm` command directly. Specify each parameter using the `--set key=value[,key=value]` argument to `helm install`. For example:
+In some cases it's more practical to provide the local configuration via the `helm` command directly. Specify each parameter using the `--set key=value[,key=value]` argument to `helm install`. These will take precedence over entries in values.yaml. For example:
```
Comment (drajen) [inline helm/charts/hpe-csi-driver/README.md]: Signify `values.yaml`?

File: helm/charts/hpe-csi-driver/README.md [hunk 12]
```diff
@@ -43,25 +42,57 @@ It's recommended to create a [values.yaml](https://github.com/hpe-storage/co-dep
...
 
 ```
-helm install --name hpe-csi hpe/hpe-csi-driver \
+# helm v3
```
Comment (drajen) [inline helm/charts/hpe-csi-driver/README.md]: Helm 3

File: helm/charts/hpe-csi-driver/README.md [hunk 13]
```diff
@@ -43,25 +42,57 @@ It's recommended to create a [values.yaml](https://github.com/hpe-storage/co-dep
...
+helm install hpe-csi hpe/hpe-csi-driver --namespace kube-system \
+--set backend=X.X.X.X --set username=admin --set password=xxxxxxxxx
+
+# helm 2.x
```
Comment (drajen) [inline helm/charts/hpe-csi-driver/README.md]: Helm 2

File: helm/charts/hpe-csi-driver/README.md [hunk 14]
```diff
@@ -43,25 +42,57 @@ It's recommended to create a [values.yaml](https://github.com/hpe-storage/co-dep
...
+
+### Upgrading the Chart
+To upgrade the chart, specify the version you want to upgrade to as below. Please do NOT re-use the values.yaml from 1.0.0 version to upgrade to later versions. Always use values.yaml from corresponding release from [values.yaml](https://github.com/hpe-storage/co-deployments/blob/master/helm/values/csi-driver)
+
+```
+# list the avaiable version of the plugin:
+helm repo update
+helm search repo hpe-csi-driver -l
+
+# select the target version to upgrade as below:
```
Comment (drajen) [inline helm/charts/hpe-csi-driver/README.md]: Select

File: helm/charts/hpe-csi-driver/README.md [hunk 15]
```diff
@@ -43,25 +42,57 @@ It's recommended to create a [values.yaml](https://github.com/hpe-storage/co-dep
...
+# helm 2.x
+# install with HPENodeInfos CRD's enabled explicitly
+helm install --name hpe-csi hpe/hpe-csi-driver --namespace kube-system -f myvalues.yaml --set crd.nodeInfo.create=true
+```
+
+### Upgrading the Chart
+To upgrade the chart, specify the version you want to upgrade to as below. Please do NOT re-use the values.yaml from 1.0.0 version to upgrade to later versions. Always use values.yaml from corresponding release from [values.yaml](https://github.com/hpe-storage/co-deployments/blob/master/helm/values/csi-driver)
+
+```
+# list the avaiable version of the plugin:
```
Comment (drajen) [inline helm/charts/hpe-csi-driver/README.md]: List

File: helm/charts/hpe-csi-driver/README.md [hunk 16]
```diff
@@ -43,25 +42,57 @@ It's recommended to create a [values.yaml](https://github.com/hpe-storage/co-dep
...
 
-**Note:** Omitting the `--name` flag will generate a human readable name.
+Install the latest chart:
+
+```
+# helm v3
+helm install hpe-csi hpe/hpe-csi-driver --namespace kube-system -f myvalues.yaml
+
+# helm 2.x
+# install with HPENodeInfos CRD's enabled explicitly
```
Comment (drajen) [inline helm/charts/hpe-csi-driver/README.md]: Install

File: helm/charts/hpe-csi-driver/README.md (hunk 5 above)
Comment (shivamerla) [reply helm/charts/hpe-csi-driver/README.md]: this is to find the available versions to install.

**Pull request merged**